grpc/interceptors.go

```diff
@@ -144,9 +164,25 @@ func (ci *clientInterceptor) intercept(
 	// Configure a grpc Trailer with respMD. This allows us to wrap error
 	// types in the server interceptor later on.
 	opts = append(opts, grpc.Trailer(&respMD))
-	err := ci.clientMetrics.UnaryClientInterceptor()(localCtx, method, req, reply, cc, invoker, opts...)
+
+	// Split the method and service name from the fullMethod.
+	// UnaryClientInterceptor's receive a `method` arg of the form
+	// "/ServiceName/MethodName"
+	service, method := splitMethodName(fullMethod)
+	// Slice the inFlightRPC inc/dec calls by method and service
+	labels := prometheus.Labels{
+		"method":  method,
+		"service": service,
+	}
+
+	// Increment the inFlightRPCs gauge for this method/service
+	ci.metrics.inFlightRPCs.With(labels).Inc()
```

Comment:
We have to decrement this on error too. Let's do this with a `defer`.

Reply:
This does get decremented on error, L186, after we unwrap the error and before returning. Do you still prefer `defer`?

Reply:
Ohh I missed that we don't return inside that `if`. Still probably makes sense to do `defer` just in case someone later adds an early return.